src/middlewared/middlewared/job.py

```diff
@@ -183,7 +183,8 @@ def __init__(self, maxlen=1000):
         self.maxlen = maxlen
         self.count = 0
         self.__dict = OrderedDict()
-        shutil.rmtree(LOGS_DIR)
+        if os.path.exists(LOGS_DIR):
```

Comment:
Looks like the most recent commit might have removed the `with contextlib` context manager?

Reply:
Oh wait, looks like this should be part of the changes for this commit?

Reply:
It would catch up as soon as the related PR is merged - a sec ; )